src/tests/ftest/aggregation/multiple_pool_cont.py

```diff
@@ -0,0 +1,195 @@
+        self.add_pool_qty(pool_qty, connect=False)
+        start_time = time.time()
+        while int(finish_time - start_time) < int(total_runtime):
+            # Since the transfer size is 1M, the objects will be inserted
+            # directly into NVMe and hence storage_index = 1
+            storage_index = 1
+
+            self.save_free_space(0, storage_index)
+
+            # Disable the aggregation
+            for pool in self.pool:
+                pool.set_property("reclaim", "disabled")
+
+            # Create the containers requested per pool
+            for pool in self.pool:
+                self.add_container_qty(container_qty, pool)
+
+            # Run ior on each container sequentially
+            for i in [1, 2]:
+                for container in self.container:
+                    ior_log = "{}_{}_{}_ior1.log".format(self.test_id,
+                                                         container.pool.uuid,
+                                                         container.uuid)
+                    try:
+                        result = run_ior(self, job_manager, ior_log, self.hostlist_clients,
+                                         self.workdir, None, self.server_group,
+                                         container.pool, container,
+                                         self.processes)
+                        self.log.info(result)
+                    except CommandFailure as error:
+                        self.log.info(error)
+                self.save_free_space(i, storage_index)
+                self.test_free_space((i-1), i)
+
+            # Enable the aggregation
+            for pool in self.pool:
+                pool.set_property("reclaim", "time")
+
+            # wait 90 seconds for files to get old enough for aggregation +
+            # 30 seconds for aggregation to start and finish
+            self.log.info("Waiting for 120 seconds for aggregation to start and finish")
+            time.sleep(120)
+            self.save_free_space(3, storage_index)
+
+            # Verify space taken by second ior is reclaimed after aggregation
+            # (logical locations will be overwritten as part of aggregation)
+            # The free space should be equal to the free space almost close
+            # to free space after 1st IOR run.
+            for pool in self.pool:
+                percentage = int((self.free_space_dict[pool.uuid][1] /
+                                  self.free_space_dict[pool.uuid][3]) * 100)
+                self.assertTrue(percentage > 97,
+                                "Aggregation did not reclaim the space")
+
+            # Destroy all containers and pools and start once again.
+            self.destroy_containers(self.container)
+            # Let the GC (garbage collection) cleanup the space.
+            time.sleep(60)
+            # Initialize the container list
+            self.container = []
+            finish_time = time.time()
```

Comment:
Do we need to log it? Since we have the total_runtime set to 900 seconds. I don't see the use of logging the difference. The while loop will anyway exit after reaching the total_runtime.

Reply:
You're right. I missed that this was used in the while loop. No need to log